system/jlib/jset.cpp

```diff
@@ -476,6 +480,17 @@ class CBitSet : public CBitSetBase<CBitSetMemoryHelper>
             memset(mem, 0, bitSetUnits*sizeof(bits_t));
         fixedMemory = true;
     }
+    CBitSet(unsigned maxBits, bool reset)
```

Comment:
@ghalliday The default function doesn't allocate memory for a given size of bitset.  From what I can see, I'd either have to pass it a MemoryBuffer (in which case it tries to deserialize from the buffer) or I'd need to pass it a pointer to a preallocated chunk of memory which means that I'd need to (1) I'd need to manage the memory externally even if it is through MemoryBuffer (2) I'd need to workout how much memory to allocate requiring the calling function to understand bitSetUnits.  It seemed cleaner to just pass the number of bits required and have CBitSet allocate and manage the required amoutn of memory.